app/upgrades/v1_5/upgrade.go

```diff
@@ -14,11 +21,57 @@ const (
 	UpgradeName = "v1.5.0"
 )
 
-func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator) upgradetypes.UpgradeHandler {
+func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator, cdc codec.Codec, storeKeys []storetypes.StoreKey, bundlesKeeper keeper.Keeper, poolKeeper *poolkeeper.Keeper) upgradetypes.UpgradeHandler {
 	return func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
-		logger := sdk.UnwrapSDKContext(ctx).Logger().With("upgrade", UpgradeName)
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+		logger := sdkCtx.Logger().With("upgrade", UpgradeName)
 		logger.Info(fmt.Sprintf("performing upgrade %v", UpgradeName))
+
+		if err := migrateStorageCosts(sdkCtx, bundlesKeeper, poolKeeper, storeKeys, cdc); err != nil {
+			return nil, err
+		}
+
 		// TODO: migrate gov params
+
 		return mm.RunMigrations(ctx, configurator, fromVM)
 	}
 }
+
+func migrateStorageCosts(sdkCtx sdk.Context, bundlesKeeper keeper.Keeper, poolKeeper *poolkeeper.Keeper, storeKeys []storetypes.StoreKey, cdc codec.Codec) error {
+	var bundlesStoreKey storetypes.StoreKey
+	for _, k := range storeKeys {
+		if k.Name() == "bundles" {
+			bundlesStoreKey = k
+			break
+		}
+	}
+	if bundlesStoreKey == nil {
+		return fmt.Errorf("store key not found: bundles")
+	}
+
+	// Get all storage providers
+	storageIds := map[uint32]struct{}{}
+	storageIds[0] = struct{}{} // Default storage provider
+	for _, pool := range poolKeeper.GetAllPools(sdkCtx) {
+		storageIds[pool.CurrentStorageProviderId] = struct{}{}
+	}
+
+	// Copy storage cost from old params to new params
+	// The storage cost of all storage providers will be the same after this migration
+	oldParams := v1_4_types.GetParams(sdkCtx, bundlesStoreKey, cdc)
+	newParams := bundlestypes.Params{
+		UploadTimeout: oldParams.UploadTimeout,
+		StorageCosts:  make([]math.LegacyDec, len(storageIds)),
+		NetworkFee:    oldParams.NetworkFee,
+		MaxPoints:     oldParams.MaxPoints,
+	}
+	for storageId := range storageIds {
+		if int(storageId) > len(newParams.StorageCosts) {
+			return fmt.Errorf("storage provider id %d is out of bounds", storageId)
```

Comment:
remove this when we turn the array into an array of {id, storageCost}

Also lets then just hardcode the value for Storage Id 1 and 2. 
A few days before finalization we will calculate the current values and use those